### ISSUE TITLE
Do controllers (in Rust) issue API requests asynchronously?

### ISSUE THREAD

**Issue by marshtompsxd** [vmware-research/verifiable-controllers#64]
TL;DR
So far all the use cases in controllers I have seen call API requests synchronously: they always immediately await after calling the async functions and there is no more than 1 pending request.

I find a few controllers from stackable, such as this zookeeper controller: https://github.com/stackabletech/zookeeper-operator. Interestingly, their controllers are built on top of another framework called operator-rs https://github.com/stackabletech/operator-rs. This is basically a wrapper around kube-rs to make it easier to build controllers in rust. This operator-rs has a set of APIs (https://github.com/stackabletech/operator-rs/blob/main/src/client.rs) to send Get/List/Create/Update/Delete requests to Kubernetes. All these APIs are implemented by calling the kube-rs APIs synchronously (immediately await after calling the async function).

I will keep investigating and checking more use cases.

**Comment by lalithsuresh** [comment]
Good find. Perhaps we can just layer on top of operator-rs if it has a more simple API boundary.
